sdk/search/Azure.Search.Documents/src/Generated/Models/SuggestResult.Serialization.cs

```diff
@@ -16,16 +16,18 @@ internal partial class SuggestResult
         internal static SuggestResult DeserializeSuggestResult(JsonElement element)
         {
             string searchtext = default;
-            IDictionary<string, object> additionalProperties = new Dictionary<string, object>();
+            IReadOnlyDictionary<string, object> additionalProperties = default;
+            Dictionary<string, object> additionalPropertiesDictionary = new Dictionary<string, object>();
             foreach (var property in element.EnumerateObject())
             {
                 if (property.NameEquals("@search.text"))
                 {
                     searchtext = property.Value.GetString();
                     continue;
                 }
-                additionalProperties.Add(property.Name, property.Value.GetObject());
+                additionalPropertiesDictionary.Add(property.Name, property.Value.GetObject());
```

Comment:
Can't they just cast back then? Seems the safer way is to create the `Dictionary<,>` first, then instantiate a `ReadOnlyDictionary` passing in that dictionary. The dictionary itself is still modifiable here, but is now encapsulated and can't be casted back to an `IDictionary`.

Reply:
Same for collections.

Reply:
We are not making defensive copies to protect from customers downcastings stuff. This collection is not shared with other objects so if someone wants to go out of the way to modify it there is nothing we can do.

Reply:
Understood, but it would be the same number of lines you have now and would make it even harder to shoot themselves in the foot (i.e. would need to use reflection to get the backing list/dictionary out).

Reply:
Same number of lines but 2x allocations.